spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2758,16 +2758,14 @@ object QueryPlanSerde extends Logging with CometExprShim {
         withInfo(join, "SortMergeJoin is not enabled")
         None
 
-      case op
-          if isCometSink(op) && op.output.forall(a =>
-            supportedDataType(
-              a.dataType,
-              // Complex type supported if
-              // - Native datafusion reader enabled (experimental) OR
-              // - conversion from Parquet/JSON enabled
-              allowComplex =
-                usingDataSourceExec(conf) || CometConf.COMET_CONVERT_FROM_PARQUET_ENABLED
-                  .get(conf) || CometConf.COMET_CONVERT_FROM_JSON_ENABLED.get(conf))) =>
```

Comment:
I don't remember why these checks were once necessary but I don't think we need them now.

Reply:
These checks were limiting our support for complex types and removing them exposed a new bug with grouping on maps, which is now fixed in this PR.